sklearn/ensemble/voting_classifier.py

```diff
@@ -19,7 +19,18 @@
 from ..base import clone
 from ..preprocessing import LabelEncoder
 from ..externals import six
-from ..utils.validation import check_is_fitted
+from ..externals.joblib import Parallel, delayed
+from ..utils.validation import has_fit_parameter, check_is_fitted
+
+
+def _parallel_fit_estimator(estimator, X, y, sample_weight):
+    """Private function used to fit an estimator within a job."""
+    if (sample_weight is not None and
+            has_fit_parameter(estimator, "sample_weight")):
```

Comment:
I'm not happy with this use of `has_fit_parameter`. So far all uses of `has_fit_parameter` raise an error when `sample_weight` isn't supported. The problem with the present use is that if `sample_weight` support is added to an estimator that formerly lacked it (something that has frequently occurred), the behaviour of `VotingClassifier` is changed without notice. I would rather, at this point, have a requirement that all estimators require `sample_weight` if `sample_weight` is provided. Another PR might consider allowing the user to specify a subset of estimators as accepting unweighted samples.